src/AppOriginal.js

```diff
@@ -0,0 +1,171 @@
+import React from "react";
+import "./App.css";
+import { makeShuffledDeck } from "./utils.js";
+
+import { Container, Row, Col } from "react-bootstrap";
+//
+
+// REFER TO IMPORT PNG IN REACT IN CHATGPT TO SOLVE IMAGES ISSUE
+
+class App extends React.Component {
+  constructor(props) {
+    // Always call super with props in constructor to initialise parent class
+    super(props);
+    this.state = {
+      // Set default value of card deck to new shuffled deck
+      cardDeck: makeShuffledDeck(),
+      // currCards holds the cards from the current round
+      currCards: [],
+      winner: null, // Add a winner state variable
+      player1Score: 0,
+      player2Score: 0,
+    };
+
+    // Define newCurrCards as a class property so it can be accessed outside dealCards()
+    this.newCurrCards = [];
+    this.player1 = {};
+    this.player2 = {};
+  }
+
+  dealCards = () => {
+    if (this.state.cardDeck.length < 2) {
+      // Check if there are enough cards in the deck
+      return;
+    }
+
+    // this.state.cardDeck.pop() modifies this.state.cardDeck array
+    this.newCurrCards = [this.state.cardDeck.pop(), this.state.cardDeck.pop()];
+    this.setState({
+      currCards: this.newCurrCards,
+      winner: null, // Resets the winner when dealing new cards
+    });
+
+    console.log(this.newCurrCards);
+  };
+
+  /////////////////////////////
+  // IMPORTANT! DETERMINES WINNER BASED ON PLAYERSCORE
+  whoWon = () => {
+    if (this.state.cardDeck.length < 2) {
+      return; // guard clause
+    }
+
+    /////////////
+    // Extract cards and determine winner
+    // extracts the two cards and saves the 2 objects into the class properties, player1 and player2
+    [this.player1, this.player2] = this.newCurrCards;
+    console.log(
+      this.player1.suit.toLowerCase(),
+      this.player2.suit.toLowerCase()
+    );
+
+    ////////////
+    // Comparing card ranks
+    let winner = null;
+    if (this.player1.rank > this.player2.rank) {
+      winner = this.player1; // player1 is declared as the winner
+
+      // Incrementing overall player1Score
+      this.setState((prevState) => ({
+        player1Score: prevState.player1Score + 1,
```

Comment:
I think you could just add the winner state update into the respective player's updates. 
```suggestion
      this.setState((prevState) => ({
        player1Score: prevState.player1Score + 1,
        winner: this.player1
```